packages/smooth_app/lib/pages/history_page.dart

```diff
@@ -0,0 +1,42 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/data_models/product_list.dart';
+import 'package:smooth_app/database/dao_product_list.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/pages/product/common/product_list_page.dart';
+
+class HistoryPage extends StatefulWidget {
+  const HistoryPage();
+  @override
+  State<HistoryPage> createState() => _HistoryPageState();
+}
+
+class _HistoryPageState extends State<HistoryPage> {
+  DaoProductList? _daoProductList;
+  final ProductList productList = ProductList(
+    listType: ProductList.LIST_TYPE_HISTORY,
+    parameters: '',
+  );
+
+  @protected
+  @mustCallSuper
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _daoProductList?.get(productList).then((_) {
+      setState(() {});
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final LocalDatabase localDatabase = context.watch<LocalDatabase>();
+    if (_daoProductList == null) {
```

Comment:
can't you move this into `didChangeDependencies` as well ? It's also called right after `initState`
`
if (_daoProductList == null) {
      _daoProductList = DaoProductList(localDatabase);
}
`

Reply:
and then we don't need the ` _daoProductList!.get(productList)...` logic here  either

Reply:
Ah, nice - I didn't know you even had access to the `context` in `didChangeDependencies`. Kinda ugly that `build` gets it as a parameter when it's already an instance variable?

Done, anyway!